### ISSUE TITLE
Multiple terraform resources are detected as multiple workspaces, resulting in inability to SSH and unrecognized name

### ISSUE THREAD

**Issue by mafredri** [coder/coder#1750]
## Steps to Reproduce

1. Uncomment the "project resource" in `examples/do-droplet-linux` (#1749)
2. Run `coder template create -d examples/do-droplet-linux`
3. Run `coder create -t do-droplet-linux my-dev`
4. Observe problems (see screenshot)

Ucommented the resource should look like:
```
resource "digitalocean_project_resources" "project" {
  project = var.step2_do_project_id
  # Workaround for terraform plan when using count.
  resources = length(digitalocean_droplet.workspace) > 0 ? [
    digitalocean_volume.home_volume.urn,
    digitalocean_droplet.workspace[0].urn
    ] : [
    digitalocean_volume.home_volume.urn
  ]
}
```

## Expected

Only one resource should be listed as the workspace.

## Actual

Multiple resources are listed and detected as workspaces, leading to unexpected behavior.

## Screenshot

<img width="695" alt="Screen Shot 2022-05-25 at 18 11 00" src="https://user-images.githubusercontent.com/147409/170319727-1b86ed81-c972-44a6-9859-6c679b4fe376.png">

**Comment by mafredri** [comment]
@kylecarbs do you think the issue in https://github.com/coder/coder/pull/1884 is related to this one?

**Comment by mafredri** [comment]
I believe this issue was fixed by #2187? // @kylecarbs

**Comment by kylecarbs** [comment]
Correct!
